File: go/enrichment/dns_tunnel.go
package main

import (
	"context"

	enpb "github.com/bearlyrunning/FindingTheNeedle/go/generated/enrichmentpb"
	spb "github.com/bearlyrunning/FindingTheNeedle/go/generated/signalpb"
)

func (dte *DNSTunnelEnricher) getName() string {
	return dte.name
}

func (dte *DNSTunnelEnricher) getSignals() []*spb.Signal {
	return dte.signals
}

func (dte *DNSTunnelEnricher) setSignals(sigs []*spb.Signal) {
	dte.signals = sigs
}

// Skipping unit test as this involves setting up mock which is out of scope for this workshop.
func (dte *DNSTunnelEnricher) enrich(ctx context.Context, c enpb.EnrichmentClient) {
	// <TODO: Implement me!>
	// Populate the `hostname` field in the spb.DNSTunnel message for all signals.
	// Hint: Check the RPC methods supported by enpb.EnrichmentClient in the generated package file.
}

File: python/enrichment/dns_tunnel.py
import enricher
import logging
from generated import enrichment_pb2 as enpb
from generated import signal_pb2 as spb

class DNSTunnelEnricher(enricher.EnricherInterface):
    def __init__(self, name="", path=""):
        self.signals = []
        self.name = name
        self.signal_path = path

    # Get enricher name.
    def getName(self) -> str:
        return self.name
    
    # Get signals associated with a given enricher.
    def getSignals(self) -> list[spb.Signal]:
        return self.signals
    
    # Set the signals to be enriched.
    def setSignals(self, signals=[]) -> None:
        self.signals = signals

    # Enrich signals based on the provided client.
    def enrich(self, client) -> None:
        # <TODO: Implement me!>
	    # Populate the `hostname` field in the spb.DNSTunnel message for all signals.
	    # Hint: Check the RPC methods supported by enpb.EnrichmentClient in the generated package file.

        # For each signal, call our IPToHost enrichment.
        for sig in self.signals:
            try:
                print("Implement me!")
            except Exception as e:
                logging.warn("failed IPToHost conversion: %s", str(e))